drivers/serial/uart_stm32.c

```diff
@@ -1429,6 +1429,7 @@ static int uart_stm32_pm_control(const struct device *dev,
 
 	/* setting a low power mode */
```

Comment:
Please, add a comment clarifying this is not setting a low power mode:

```suggestion
	/* System can not enter low power modes if UART is still busy,
	  * e.g., transmitting. The implementation below blocks until this
	  * condition is fulfilled, however, no actual low power mode is set
	  * for the UART peripheral.
	  */
```